src/Spice86.Shared/Rgb.cs

```diff
@@ -4,25 +4,78 @@
 /// RGB representation of a color.
 /// </summary>
 public class Rgb {
-    public byte R { get; set; }
+    /// <summary>
+    /// Red channel, on 8 bit
+    /// </summary>
+    public byte R8 { get; set; }
 
-    public byte G { get; set; }
+    /// <summary>
+    /// Green channel, on 8 bit
+    /// </summary>
+    public byte G8 { get; set; }
 
-    public byte B { get; set; }
+    /// <summary>
+    /// Blue channel, on 8 bit
+    /// </summary>
+    public byte B8 { get; set; }
+    
+    
+    /// <summary>
+    /// Red channel, on 6 bit
+    /// </summary>
+    public byte R6 { get; set; }
 
-    public uint ToRgba() {
-        return (uint)(R << 16 | G << 8 | B) | 0xFF000000;
-    }
+    /// <summary>
+    /// Green channel, on 6 bit
+    /// </summary>
+    public byte G6 { get; set; }
 
-    public uint ToBgra() {
-        return (uint)(B << 16 | G << 8 | R) | 0xFF000000;
-    }
+    /// <summary>
+    /// Blue channel, on 6 bit
+    /// </summary>
+    public byte B6 { get; set; }
 
-    public uint ToArgb() {
-        return 0xFF000000 | (uint)R << 16 | (uint)G << 8 | B;
-    }
+    public uint ToRgba() => 0x000000FF | ToRgb();
```

Comment:
I think the ToRgb needs to be shifted left 8 bits.
Or `R8 << 24 | G8 << 16 | B8 << 8 | 0xFF`

Reply:
Change applied. Thanks !